src/models/petsModel.js

```diff
@@ -49,7 +53,80 @@ const createNewPet = async (newPet) => {
   });
 }
 
+const getPetById = async (db, petID) => {
+  const sql = 'SELECT p.*, s.ShelterName, s.Address, s.EmailAddress, s.PhoneNumber, s.Website '
+            + 'FROM Pet p INNER JOIN Shelter s ON p.ShelterID = s.ShelterID '
+            + 'WHERE PetID = ?';
+  const values = [petID];
+  log.debug("Running getPetById sql = " + mysql.format(sql, values));
+  return new Promise((resolve, reject) => {
+    db.query(sql, values, (err, res, fields) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    });
+  });
+};
+
+const deletePetById = async (db, petID) => {
+  const sql = 'DELETE FROM Pet WHERE PetID = ?';
+  const values = [petID];
+  log.debug("Running deletePetById sql = " + mysql.format(sql, values));
+  return new Promise((resolve, reject) => {
+    db.query(sql, values, (err, res, fields) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    });
+  });
+};
+
+const updatePetById = async (db, petID, petToUpdate) => {
```

Comment:
What happens if all JSON attributes are not sent? We want to make sure to only update the values that are sent in the request body

Reply:
What I wanted to do with the frontend is that I would load the current data into the inputs, send all attributes in the request body, and the API would just update all attributes and doesn't have to know if an attribute has been updated.

Reply:
That's a good idea for the frontend. I don't think this PATCH request is consistent with our other PATCH requests though. If we want to require all attributes, we should make this a PUT request. I personally think that we should allow a request to only send some attributes. It may be useful later on

Reply:
Yeah I think you're right about it being a PUT request. I'll change my PATCH method to be consistent with the other PATCH requests.